.github/workflows/core.yml

```diff
@@ -89,3 +89,17 @@ jobs:
       run: scripts/setup_bundler.sh
     - name: PodLibLint Core Cron
       run: scripts/third_party/travis/retry.sh scripts/pod_lib_lint.rb FirebaseCore.podspec --platforms=${{ matrix.target }} ${{ matrix.flags }}
+
+  podspec-presubmit:
+    # Don't run on private repo unless it is a PR.
+    if: github.repository == 'Firebase/firebase-ios-sdk' && github.event.pull_request.merged != true
+    runs-on: macOS-latest
+    strategy:
+      matrix:
+        target: [ios, tvos, macos, watchos]
```

Comment:
For `pod spec lint` tests we could skip the matrix and the `--platforms` option and just always do the default of all platforms specified in the podspec.